includes/ingest.batch.inc

```diff
@@ -410,31 +409,45 @@ function islandora_batch_process_results($results, $timeout, $lock_timeout, &$co
         // XXX: Due to how the things currently work, the user name and
         // password hash gets serialized, so later changes to the password
         // can break things... Let's set the password again.
-        $username = $ingest_object->repository->api->connection->username;
+        $username = $islandora_batch_object->repository->api->connection->username;
         $user = user_load_by_name($username);
-        if ($ingest_object->repository->api->connection->password != $user->pass) {
-          $ingest_object->repository->api->connection->password = $user->pass;
+        if ($islandora_batch_object->repository->api->connection->password != $user->pass) {
+          $islandora_batch_object->repository->api->connection->password = $user->pass;
         }
 
         // Push to backend.
-        $ingested_object = islandora_add_object($ingest_object);
-        if ($ingested_object) {
-          $context['message'] = t('Ingested %pid.', array('%pid' => $ingested_object->id));
+        $actionresult_object = $islandora_batch_object->batchRepositoryAction();
+
+        if ($actionresult_object) {
+          $context['message'] = t('Successfully %action %pid.',
+            array(
+              '%pid' => $actionresult_object->id,
+              '%action' => $islandora_batch_object->getRepositoryActionDescription(),
+            ));
         }
         else {
-          // Failed to ingest...  Flag an error.
+          // Failed to process on backend...  Flag an error.
           $object['state'] = ISLANDORA_BATCH_STATE__ERROR;
-          $context['message'] = t('Unknown error: Failed to ingest %pid.', array('%pid' => $ingest_object->id));
+          $context['message'] = t('Unknown error: Failed to %action %pid.',
+            array(
+              '%pid' => $islandora_batch_object->id,
+              '%action' => $islandora_batch_object->getRepositoryActionDescription(),
+            ));
         }
       }
       catch (Exception $e) {
-        // Failed to ingest...  Flag an error.
+        // Backend exception happened...  Flag an error.
         $object['state'] = ISLANDORA_BATCH_STATE__ERROR;
-        $context['message'] = t('Exception occured: Failed to ingest %pid.', array('%pid' => $ingest_object->id));
+        $context['message'] = t('Exception occured: Failed to %action %pid because of %message.',
+        array(
+          '%pid' => $islandora_batch_object->id,
+          '%action' => $islandora_batch_object->getRepositoryActionDescription(),
+          '%message' => $e->getMessage(),
+        ));
       }
     }
     else {
-      $context['message'] = t('%pid not ready for ingest.', array('%pid' => $ingest_object->id));
+      $context['message'] = t('%pid not ready for ingest.', array('%pid' => $islandora_batch_object->id));
```

Comment:
Not sure, but should `ingest` not be replaced by the selected action (if so, perhaps re-wording to `'%pid cannot be %action: object not ready.'` could help with the grammar...)?

Reply:
Well, I think the more compact the message is, the better. On the other hand, leaving past tenses lying around is somehow awkward (by the way, I guess it would be a good idea to wrap whatever we decide on with `t()`) - let alone present + past tenses. If you do not like my [earlier suggestion](https://github.com/Islandora/islandora_batch/pull/105#discussion_r214992236) of reformulating that one message into past tense, how about returning `t("Ingest")` and alerting the user with `'%action action for %pid succeeded.'`, `'%action action for %pid failed.'`, etc... (or similar)?

Regarding the naming of functions (and files?), perhaps a little bit of refactoring would be good. I think, otherwise, this might cause confusion if picked up after some time. But feel free (you, or anybody) to disagree...

Reply:
Ok, I missed your earlier suggestion! (sorry), yeah, sounds fine to me and i will go for that, thanks. t() and watchdog  don't play well... since we are kinda accumulating the messages. I remember having quite some issues with that in the past so will stay away from that now. Also, i always forget our classes are really not generic php classes, they are used inside the weirdness of drupal, so t() can be used... 

About naming of functions and files. Refactor can bring fear and riots as said before. Will wait a bit and if someone else feels it is needed right now can do of course. I would need to deprecate some functions and maybe too much. Lets discuss this. This was meant to actually be eventually merged!